src/Component.php

```diff
@@ -20,16 +20,8 @@ public function run(): void
             /** @var Config $config */
             $config = $this->getConfig();
             $client = new Client(['token' => $config->getToken(), 'url' => $config->getUrl()]);
-            $tokenInfo = $client->verifyToken();
-            if (count($tokenInfo['bucketPermissions']) > 1) {
-                throw new UserException('The token must have read-only permissions to a single bucket only.');
-            }
-            $bucket = array_keys($tokenInfo['bucketPermissions'])[0];
-            if ($tokenInfo['bucketPermissions'][$bucket] !== 'read') {
-                throw new UserException(
-                    sprintf('The token must have read-only permissions to the bucket "%s".', $bucket)
-                );
-            }
+            $tokenValidator = new TokenValidator($client);
+            $bucket = $tokenValidator->validateBucket();
```

Comment:
Tady se mi nelibi nazev tridy a metody, ktery se o to stara. Podle nazvu bych rekl, ze je to void, ale ono to vraci string. Byl bych spise pro metodu (klidne rovnou v tady napr. `obtainBucket`), ktere predas `$tokenInfo['bucketPermissions']` a ona se postara o to, ze kdyz je tam prave jeden, tak ti vrati jeho string.

Reply:
no jmeno je divny no, `obtainBucket` to me zase zni skoro jako `getBucket` a u toho necekas, ze to vyhazuje validacni errory :)